core/src/cluster_info.rs

```diff
@@ -282,7 +282,7 @@ impl ClusterInfo {
 
     pub fn push_vote(&mut self, vote: Transaction) {
         let now = timestamp();
-        let vote = Vote::new(vote, now);
+        let vote = Vote::new(&self.id(), vote, now);
```

Comment:
"Voter" pubkey here being the vote account_id? Because Fullnode may not own that keypair and it would be sort of wasteful to make another request to sign this CrdsValue::Vote with that keypair. 

Instead this falls nicely in line with the way ContactInfo is gossiped. Essentially a node that puts data out into the gossip network _must_ sign that data with a keypair that it owns. All gossiped values are signed by the peer that generated the message and it's easy to verify that.

The Vote itself will be verified by system program. This allows us to easily package wallclock into the crds vote.

Reply:
@sagar-solana Transaction.signatures[0] would be the signature.  We just need to translate last_id into a wallclock.

Reply:
Yup but @rob-solana and I decided it was best to not try and tie in external state information (like blockhashes -> timestamp) into gossip. It was far more trivial to sign these messages with the node's keypair (used for signing blobs and other gossip messages anyway).

Reply:
@rob-solana any reason besides expendience?

Reply:
main reason was to try to keep gossip as dumb (i.e. simple) as possible (not needing to know how to unpack a TX, e.g.)
it'd be more expedient to just hack into a vote